### PR TITLE
Add automatic doc generation using html_of_wiki and Travis.

### PR DESCRIPTION
**Pull request by leovalais**
This installation replaces the old `html_of_wiki` workflow. I created an orphan branch `wikidoc` which is to contain the up-to-date documentation. On each commit (classic or merge) on this branch, the CI will run, trigger the new version of `html_of_wiki` and compile the documentation. Then, only on `wikidoc` (not on forked branches), if the website generation has succeeded, Travis will push the HTML pages automatically on branch `gh-pages`, thus updating the website.

I'll write a section on ocsigen.org about how to update the documentation for a new release. 
In short, on branch `wikidoc`, copy the `dev` directory and rename it with the new version number, commit, push and the CI will regenerate the website automatically. The `dev` directory should be kept up to date. To do that, as before, manually generate the api wikis using `ocamldoc` and replace the old `dev/api` on `wikidoc`. Then commit push the new dev api and the corresponding doc will also be updated automatically.

The files to review are `.travis.yml`, `.howdocgen` and the global architecture.

### CODE REVIEW

**Comment by aantron** [comment]
Please let me know if/when action by me is needed.

**Comment by balat** [comment]
@aantron I will merge this PR (as we already did for all other projects).
This will trigger the regeneration of the doc from the new wikidoc branch and update the Web site.

**Comment by hcarty** [comment]
Are the docs available now?  The link from main Lwt repo description leads to a 404.

**Comment by leovalais** [comment]
@hcarty thanks, #651 should fix that.
